doc/kubernetes/modules/ROOT/pages/running/keycloak-with-external-infinispan-deployment.adoc

```diff
@@ -0,0 +1,62 @@
+= Keycloak with external Infinispan Deployment
+:description: This describes configuration tweaks and changes on how to run Keycloak with an external Infinispan Deployment under load.
+
+{description}
+
+== Audience
+
+This guide describes advanced Infinispan configurations for Keycloak on Kubernetes which are load tested and will recover from single Pod failures.
+
+While the Helm charts in the Keycloak Benchmark project mix different aspects of production style deployments with instrumentation and monitoring, this documentation focuses on a minimal deployment with optional add-ons which admins can opt in for their own deployments.
+
+See xref:running/index.adoc[] for additional guides.
+
+== Prerequisites
+
+* OpenShift or Kubernetes cluster running.
+* Understanding of the xref:running/keycloak-deployment.adoc[Basic Keycloak deployment].
+* Understanding of an https://infinispan.org/docs/infinispan-operator/main/operator.html[Infinispan deployment using its Operator].
+
+== Procedure
+
+* Prepare an Infinispan Cache configuration XML, which would then be consumed in the Keycloak CR.
+* Create Config Map, Secret and other Kubernetes resources needed for Keycloak to connect to an external Infinispan deployment.
+* Extend the Keycloak Custom Resource with below Kubernetes deployment resources.
+
+=== Remote store username and password defined in a Kubernetes Secret
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: remote-store-secret
+  namespace: keycloak
+type: Opaque
+data:
+  username: <base-64 encoded String>
+  password: <base-64 encoded String>
+----
+
+=== Keycloak Cache configuration defined in a Kubernetes Config Map
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kcb-infinispan-cache-config
+  namespace: keycloak
+data: {}
+----
+
+=== Keycloak CR with external Infinispan updates
+[source,yaml]
+----
+include::example$helm/keycloak-ispn.yaml[tag=keycloak-ispn]
```

Comment:
I see that this is missing some options that are present in https://www.keycloak.org/keycloak-benchmark/kubernetes-guide/latest/running/keycloak-deployment#procedure for example database config, should we include it too? Or we rely on the prerequisite to understand the previous guide.

Reply:
right we are treating this as an incremental update to the basic Keycloak Deployment.